### ISSUE TITLE
System.IO.Packaging.dll is not needed in System.IO.Packaging package for net462 target

### ISSUE THREAD

**Issue by djohnst10** [dotnet/runtime#105896]
We have a .NET Framework library project that targets .NET Framework 4.7.2 and uses the System.IO.Packaging APIs. We have now converted this project to an SDK project that multi-targets net472 and net8. During the conversion I added a PackageReference for the System.IO.Packaging NuGet package. The System.IO.Packaging NuGet package version 8.0.0 targets net6, net7, net8, netstandard2.0, and net462.

I had assumed that when building our library's net472 version that the System.IO.Packaging NuGet package would just use a framework reference to WindowsBase, where the System.IO.Packaging code lives.

That isn't exactly what happens. The NuGet package does add a framework reference to WindowsBase, but it also includes a System.IO.Packaging.dll file when doing a .NET Framework build and when publishing the .NET Framework executable that references the library.

When I investigated why System.IO.Packaging.dll was needed, it turns out that it was not needed.

After loading the compiled library in ILSpy, I saw that our library does not have a reference to the System.IO.Packaging assembly, only the WindowsBase assembly. Looking at the System.IO.Packaging assembly in ILSpy shows that none of its types are public - all are internal. Many of the types are Attribute types like AllowNullAttribute or NotNullAttribute and resource string wrapper types like System.SR or FxResources.System.IO.Packaging.SR. Because none of the types in System.IO.Packaging.dll are used by our library, the C# compiler does not include a reference to that assembly in the metadata of our compiled library assembly that targets net472.

It seems like these types were used by code that was not included in the compilation for the System.IO.Packaging NuGet package's net462 target. Looking at the assemblies built for net6, net7, net8 and netstandard2.0 shows that they have much more of the System.IO.Packaging types that use the nullability attributes and resource string wrapper types.

Bottom line: there should be no System.IO.Packaging.dll included in the System.IO.Packaging NuGet package for the net462 target.

**Comment by huoyaoyuan** [comment]
It's for .NET Standard support.

When decompiling the dll, note the `TypeForwardedTo` attributes:

![image](https://github.com/user-attachments/assets/fa8aba36-b91d-493d-8976-208c7acd8d2b)

If you click the type names, you will be redirected to the corresponding implementation in `WindowsBase`.

Assembly name is a part of a type's identity. An assembly targeting .NET Standard will reference `[System.IO.Packaging]System.IO.Packaging.ZipPackage`, the type forwarding attribute redirects the identity to `[WindowsBase]System.IO.Packaging.ZipPackage`.

For projects multi-targeting .NET Framework and .NET Core, the package is unnecessary since it's inbox on both frameworks. It's only required for projects targeting .NET Standard.

**Comment by ericstj** [comment]
@huoyaoyuan has it right.  The assembly is required.  It's for the case where you consume some library that targeted `netstandard2.0` and reference the package.  If all your libraries target .NETFramework you don't need the package at all.
